src/components/dataset/new.vue

```diff
@@ -0,0 +1,152 @@
+              {{ $t('action.create') }} <spinner :state="awaitingResponse"/>
+            </button>
+            <button type="button" class="btn btn-link"
+              :aria-disabled="awaitingResponse" @click="hideOrComplete">
+              {{ $t('action.cancel') }}
+            </button>
+          </div>
+        </form>
+      </template>
+      <template v-else>
+        <div id="entity-list-new-success" class="modal-introduction">
+          <p>
+            <span class="icon-check-circle"></span>
+            <strong>{{ $t('common.success') }}</strong>
+            <sentence-separator/>
+            <span>{{ $t('success[0]', createdDataset) }}</span>
+          </p>
+          <p>{{ $t('success[1]') }}</p>
+        </div>
+        <div class="modal-actions">
+          <button type="button" class="btn btn-primary" @click="complete">
+            {{ $t('action.done') }}
+          </button>
+        </div>
+      </template>
+    </template>
+  </modal>
+</template>
+
+<script setup>
+import { ref, watch } from 'vue';
+
+import Modal from '../modal.vue';
+import DocLink from '../doc-link.vue';
+import FormGroup from '../form-group.vue';
+import SentenceSeparator from '../sentence-separator.vue';
+import Spinner from '../spinner.vue';
+import useRequest from '../../composables/request';
+import { apiPaths } from '../../util/request';
+import { useRequestData } from '../../request-data';
+import { noop } from '../../util/util';
+
+const { request, awaitingResponse } = useRequest();
+const { project } = useRequestData();
+
+defineOptions({
+  name: 'DatasetNew'
+});
+const props = defineProps({
+  state: {
+    type: Boolean,
+    default: false
+  }
+});
+const name = ref('');
+const step = ref(0);
+const createdDataset = ref(null);
+
+const emit = defineEmits(['hide', 'success']);
+
+watch(() => props.state, (state) => {
+  if (!state) name.value = '';
+});
+
+const submit = () => {
+  request({
+    method: 'POST',
+    url: apiPaths.datasets(project.id),
+    data: { name: name.value }
+  })
+    .then(({ data }) => {
+      // Reset the form
+      name.value = '';
+      step.value = 1;
+      createdDataset.value = data;
+    })
+    .catch(noop);
+};
+
+const complete = () => {
+  emit('success', createdDataset.value);
+};
+
+const hideOrComplete = () => {
+  if (createdDataset.value == null)
+    emit('hide');
+  else
+    complete();
+};
+</script>
+
+<style lang="scss">
+@import '../../assets/scss/variables';
+
+#entity-list-new-success {
+  .icon-check-circle {
+    color: $color-success;
+    font-size: 32px;
+    margin-right: 6px;
+    vertical-align: middle;
+  }
+}
+</style>
+
+<i18n lang="json5">
+{
+  "en": {
+    // This is the title at the top of a pop-up.
+    "title": "Create Entity List",
+    "introduction": [
+      "Entities let you share information between Forms so you can collect longitudinal data, manage cases over time, and represent other workflows with multiple steps."
```

Comment:
Ask issa or Aly what this text should say.
If keeping this text, use the transifex key to reuse existing translation.